agent-manager/src/agent_manager/agent_manager.py

```diff
@@ -41,11 +41,29 @@ def connect_to_llama_stack(self):
         else:
             logging.debug("Already connected to LlamaStack")
 
+    def register_mcp_toolgroups(self, agent):
```

Comment:
Based on the pattern @tchughesiv introduced mcp servers should likely be a new type of asset.

Reply:
thinking more, it may make more sense to have each MCP server self-register with llamastack as they start up

Reply:
Also I guess that then begs the question of how do we avoid re-registration when pods restart etc. Ideally the registration would be done when the service starts ?

Reply:
With a bit more reading an init container may not be what we want as it sounds like it will run for every intance in the pod